### PR TITLE
feat(api-client): Legal hold endpoints

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: packages/api-client/src/main/team/member/MemberAPI.ts [hunk 1]
```diff
@@ -85,6 +87,45 @@ class MemberAPI {
...
+    return this.client.sendJSON<void>(config).then(response => response.data);
+  }
+
+  public deleteMemberLegalHold(teamId: string, userId: string): Promise<void> {
+    const config: AxiosRequestConfig = {
+      method: 'delete',
+      url: `${TeamAPI.URL.TEAMS}/${teamId}/${MemberAPI.URL.LEGAL_HOLD}/${userId}`,
+    };
+
+    return this.client.sendJSON<void>(config).then(response => response.data);
```
Comment (ffflorian) [inline packages/api-client/src/main/team/member/MemberAPI.ts]:
```suggestion
    await this.client.sendJSON<void>(config);
```
And the same for the other functions if they don't return anything.

File: packages/api-client/src/main/client/RegisteredClient.ts [hunk 2]
```diff
@@ -17,16 +17,20 @@
...
 
-interface RegisteredClient {
-  address?: string; // IP address
-  class: 'desktop' | 'phone' | 'tablet';
-  cookie: string; // Cookie label
-  id: string; // Client ID
+import {ClientClassification} from './ClientClassification';
+
+export interface RegisteredClient {
+  /** The IP address */
```
Comment (bennyn) [inline packages/api-client/src/main/client/RegisteredClient.ts]: Let's specify that it is the IP address from where the client has been registered.

File: packages/api-client/src/main/team/member/MemberAPI.ts [hunk 3]
```diff
@@ -26,9 +26,13 @@ import {TeamAPI} from '../team/TeamAPI';
...
-  static URL = {
-    MEMBERS: 'members',
-  };
+  static get URL() {
+    return {
+      APPROVE_LEGAL_HOLD: 'approve',
+      LEGAL_HOLD: 'legalhold',
+      MEMBERS: 'members',
+    };
+  }
```
Comment (ffflorian) [inline packages/api-client/src/main/team/member/MemberAPI.ts]: Let's keep this as a const

File: packages/api-client/src/main/team/legalhold/LegalHoldData.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+export enum LegalHoldStatus {
+  ENABLED_NOT_CONFIGURED = 'enabled_not_configured',
+  ENABLED_CONFIGURED = 'enabled_configured',
+  DISABLED = 'disabled',
+}
+
+export interface LegalHoldData {
+  enabled: LegalHoldStatus;
+  // settings data is only present when legal hold is LegalHoldStatus.ENABLED_CONFIGURED
```
Comment (bennyn) [inline packages/api-client/src/main/team/legalhold/LegalHoldData.ts]: Can we express this behaviour with TypeScript?

File: packages/api-client/src/main/team/legalhold/LegalHoldData.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  settings?: {
+    team_id: string;
+    fingerprint: string;
```
Comment (bennyn) [inline packages/api-client/src/main/team/legalhold/LegalHoldData.ts]: Sorting

File: packages/api-client/src/main/team/legalhold/LegalHoldAPI.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  public async deleteMemberLegalHold(teamId: string, userId: string): Promise<void> {
+    const config: AxiosRequestConfig = {
+      method: 'delete',
+      url: `${TeamAPI.URL.TEAMS}/${teamId}/${LegalHoldAPI.URL.LEGAL_HOLD}/${userId}`,
+    };
+
+    await this.client.sendJSON<void>(config);
+  }
+
+  public async postMemberLegalHold(teamId: string, userId: string): Promise<void> {
```
Comment (bennyn) [inline packages/api-client/src/main/team/legalhold/LegalHoldAPI.ts]: There is `postMemberLegalHold` and `getLegalHoldSettings`. 

We should have a uniform naming where it is either:
- postMemberLegalHold
- getSettingsLegalHold

Or
- postLegalHoldMember
- getLegalHoldSettings

File: packages/api-client/src/main/team/legalhold/LegalHoldData.ts (hunk 4 above)
Comment (ffflorian) [reply packages/api-client/src/main/team/legalhold/LegalHoldData.ts]: Yes I already have an idea :)

Comment (bennyn) [reply packages/api-client/src/main/team/legalhold/LegalHoldData.ts]: Great. Go for it! :D

Comment (ffflorian) [reply packages/api-client/src/main/team/legalhold/LegalHoldData.ts]: Done with 2388c5b30f7469833cb23318c1c105f58934483a

**Pull request merged**